Mathlib/MeasureTheory/Group/ModularCharacter.lean

```diff
@@ -0,0 +1,125 @@
+      ∃ f : C(G, ℝ), HasCompactSupport f ∧ 0 ≤ f ∧ f 1 ≠ 0 := exists_continuous_nonneg_pos 1
+    have int_f_ne_zero (μ₀ : Measure G) [IsHaarMeasure μ₀] : ∫ x, f x ∂μ₀ ≠ 0 :=
+    ne_of_gt (f_cont.integral_pos_of_hasCompactSupport_nonneg_nonzero f_comp f_nonneg f_one)
+    apply NNReal.coe_injective
+    have t : (∫ x, f (x * g) ∂ν) = (∫ x, f (x * g) ∂(haarScalarFactor ν μ • μ)) := by
+      refine integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport ν μ ?_ ?_
+      · exact Continuous.comp' f_cont (continuous_mul_right g)
+      · have j : (fun x ↦ f (x * g)) = (f ∘ (Homeomorph.mulRight g)) := rfl
+        rw [j]
+        exact HasCompactSupport.comp_homeomorph f_comp _
+    have r : (haarScalarFactor ν μ : ℝ) / (haarScalarFactor ν μ) = 1 := by
+      refine div_self ?_
+      rw [NNReal.coe_ne_zero]
+      apply (ne_of_lt (haarScalarFactor_pos_of_isHaarMeasure _ _)).symm
+    calc
+    ↑(modularCharacterFun g) = ↑(haarScalarFactor (map (· * g) ν) ν) := rfl
+    _ = (∫ x, f x ∂(map (· * g) ν)) / ∫ x, f x ∂ν :=
+      haarScalarFactor_eq_integral_div _ _ f_cont f_comp (int_f_ne_zero ν)
+    _ = (∫ x, f (x * g) ∂ν) / ∫ x, f x ∂ν := by
+      rw [integral_map (AEMeasurable.mul_const aemeasurable_id' _)
+      (Continuous.aestronglyMeasurable f_cont)]
+    _ = (∫ x, f (x * g) ∂(haarScalarFactor ν μ • μ)) / ∫ x, f x ∂ν := by rw [t]
+    _ = (∫ x, f (x * g) ∂(haarScalarFactor ν μ • μ)) / ∫ x, f x ∂(haarScalarFactor ν μ • μ) := by
+      rw [integral_isMulLeftInvariant_eq_smul_of_hasCompactSupport ν μ f_cont f_comp]
+    _ = (haarScalarFactor ν μ • ∫ x, f (x * g) ∂μ) / (haarScalarFactor ν μ • ∫ x, f x ∂μ) := by
+      rw [integral_smul_nnreal_measure, integral_smul_nnreal_measure]
+    _ = (haarScalarFactor ν μ / haarScalarFactor ν μ) * ((∫ x, f (x * g) ∂μ) / ∫ x, f x ∂μ) :=
+      mul_div_mul_comm _ _ _ _
+    _ = 1 * ((∫ x, f (x * g) ∂μ) / ∫ x, f x ∂μ) := by rw [r]
+    _ = (∫ x, f (x * g) ∂μ) / ∫ x, f x ∂μ := by rw [one_mul]
+    _ = (∫ x, f x ∂(map (· * g) μ)) / ∫ x, f x ∂μ := by
+      rw [integral_map (AEMeasurable.mul_const aemeasurable_id' _)
+      (Continuous.aestronglyMeasurable f_cont)]
+    _ = haarScalarFactor (map (· * g) μ) μ :=
+      (haarScalarFactor_eq_integral_div _ _ f_cont f_comp (int_f_ne_zero μ)).symm
+
+@[to_additive map_right_add_eq_addModularCharacterFun_smul]
+lemma map_right_mul_eq_modularCharacterFun_smul (μ : Measure G) [IsHaarMeasure μ] [InnerRegular μ]
+    (g : G) : (map (· * g) μ) = modularCharacterFun g • μ := by
+    rw [modularCharacterFun_eq_haarScalarFactor μ _]
+    exact isMulLeftInvariant_eq_smul_of_innerRegular _ μ
+
+@[to_additive addModularCharacter_pos]
+lemma modularCharacter_pos (μ : Measure G) [IsHaarMeasure μ] (g : G) :
+    0 < modularCharacterFun g := by
+      rw [modularCharacterFun_eq_haarScalarFactor μ _]
+      exact haarScalarFactor_pos_of_isHaarMeasure _ μ
+
+/-- The modular character homomorphism. -/
+noncomputable def modularCharacter:
```

Comment:
I couldn't add `to_additive` here because I think he tries to turn the multiplication on RR to addition too? I would appreciate help with this.

Reply:
[It seems like there isn't much to do about it.](https://leanprover.zulipchat.com/#narrow/channel/113489-new-members/topic/Homomorphisms.20from.20additive.20group.20to.20multiplicative)
I think for now I won't define the additive modular character, but maybe in a future PR.